sherpa/models/model.py

```diff
@@ -473,12 +475,10 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
         vals = func(cls, pars, xlo, *args, **kwargs)
 
         # remove first item in queue and remove from cache
-        queue = cls._queue
-        key = queue.pop(0)
-        cache.pop(key, None)
+        # if cache is reaching max size.
```

Comment:
```suggestion
        # if the cache is full.
```

We **could** change from a `list` to a [deque](https://docs.python.org/3/library/collections.html#deque-objects) object. I have seen some complaints about the  runtime speed of `deque` but have not investigated enough to know if they are meaningful or not. We are also not really doing much with the queue so is the semantic improvement worth it?

Reply:
In the PR, I use a dict. Python spent a lot of effort recently to improve performance of dict AND they also maintain the insertion order (since 3.9 or something like that). With that, there really is not much difference between a dict and a `deque` any more in the way we use it here, so I propose to stick with the more-known `dict` instead of the `deque` (performance aside).